File: leetcode-js/179-largest-number.js
/**
 * @param {number[]} nums
 * @return {string}
 */
var largestNumber = function(nums) {
    if (!nums.length) {
        return '';
    }

    const snums = nums.map((n) => n.toString()).sort((a, b) => {
        const s1 = a + b;
        const s2 = b + a;
        return s1.localeCompare(s2);
    });

    if (snums[snums.length - 1][0] === '0') {
        return '0';
    }

    let result = '';

    for (const s of snums) {
        result = s + result;
    }

    return result;
};

File: leetcode-java/179-largest-number.java
public class Solution {
    public String largestNumber(int[] nums) {
        if (nums.length == 0) {
            return "";
        }

        String[] strs = new String[nums.length];

        for (int i = 0; i < nums.length; i++) {
            strs[i] = String.valueOf(nums[i]);
        }

        Comparator<String> comp = new Comparator<String>() {
            @Override
            public int compare(String a, String b) {
                String s1 = a + b;
                String s2 = b + a;
                return s2.compareTo(s1);
            }
        };

        Arrays.sort(strs, comp);

        if (strs[0].charAt(0) == '0') {
            return "0";
        }

        StringBuilder builder = new StringBuilder();

        for (String s : strs) {
            builder.append(s);
        }

        return builder.toString();
    }
}
